Source/drlg_l1.cpp

```diff
@@ -2048,30 +2038,50 @@ void CryptFloor(int rndper)
 bool PlaceCathedralStairs(lvl_entry entry)
 {
 	bool success = true;
+	Point position;
+	Displacement offset { 3, 4 };
 
 	// Place poison water entrance
 	if (Quests[Q_PWATER].IsAvailable()) {
-		if (!PlaceMiniSet(PWATERIN, entry == ENTRY_RTNLVL))
+		Point miniPosition;
+		if (!PlaceMiniSet(PWATERIN, &miniPosition)) {
 			success = false;
-		if (entry == ENTRY_RTNLVL)
-			ViewPosition += Displacement { 2, 3 };
+		} else {

```

Comment:
Returning optional would allow something like:

```cpp
optional<Point> position = PlaceMiniSet(PWATERIN);
if (position) {
    Point miniPosition = position.get();
```